### PR TITLE
Update ecmaVersion to latest for optional chaining

### PR DESCRIPTION
**Pull request by kailin0512**

### CODE REVIEW

**Review by mikewoo200 — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ module.exports = {
         'eslint-config-ebay/rules/es6'
     ].map(require.resolve),
     parserOptions: {
-        ecmaVersion: 2017
+        ecmaVersion: 2020
```
Comment (mikewoo200) [inline index.js]: New version in package.json?

**Comment by DylanPiercey** [comment]
@kailin0512 this config doesn't reflect what we recommended internally. Should we deprecate this? If not at the very least we should use "latest" instead of "2022" so that you don't need to make this PR every year 😅

**Comment by kailin0512** [comment]
> @kailin0512 this config doesn't reflect what we recommended internally. Should we deprecate this? If not at the very least we should use "latest" instead of "2022" so that you don't need to make this PR every year 😅

@DylanPiercey I updated to latest which should be the default value. Initially I thought we override this to 2017 for some specific reason but if you think always pull the latest should be fine we can do that as well. I was just encounter the optional chaining problem so I updated to minimum supported version. 

I am ok to deprecate this as well but when did we decide not to recommend that internally? At least for couple platform modules and sample service it is still extending this module.

**Review by ianmcburnie — APPROVED** [review]

**Review by DylanPiercey — APPROVED** [review]

**Comment by ianmcburnie** [comment]
Will merge. Then I will bump the package version (new minor version), commit that change and publish to NPM.

**Pull request merged**